UnityProject/Assets/Scripts/Systems/Inventory/Pickupable.cs

```diff
@@ -224,7 +225,7 @@ private void RpcPickupAnimation(GameObject interactor)
 		if (interactor == null) return;
 
 		PickupAnim(interactor);
-		OnMoveToPlayerInventory?.Invoke(interactor);
+		if(CustomNetworkManager.IsServer == false) OnMoveToPlayerInventory?.Invoke(interactor);
```

Comment:
why is this event invocation needed? You are already calling it in the main coroutine (which is the one that should be tasked with invoking). I remember I told you in another review to remove the invocation from the animation method.

Reply:
One is called on the server and the other called on the client
`[ClientRpc]` and `ServerPerfomInteraction()`

Reply:
the reason there is a if to check if its the server because in the editor it can trigger twice

Reply:
it does feel hacky to use the animation RPC method to invoke the event on client, and I thought ``ServerPerformInteraction`` actually runs on both sides. Let me ask around in #general-dev